core/multicluster/middleware.go

```diff
@@ -0,0 +1,50 @@
+package multicluster
```

Comment:
is there any way that we can "hide" this a bit better?

given that core is technically not supposed to have a feature called multicluster, it may look sus when people read the code.

perhaps we can rename all this to `clustersmanager` or something?

Reply:
we'll not hide that enterprise has that feature, so that's not an issue, although having a `SingleClusterFetcher` inside `multicluster` package looks weird. 😄  I'll change it.